src/main/java/com/starbucks/service/OrderService.java

```diff
@@ -17,40 +20,44 @@ public OrderService(OrderRepository orderRepository) {
         this.orderRepository = orderRepository;
     }
 
-    public String saveOrder(Order order) {
-
-        orderRepository.save ( order );
-        return order.toString ( );
+    public Order saveOrder(Order order) {
+        return orderRepository.save(order);
     }
 
-    public String updateOrder(Order order , UUID id) {
-
-        if (orderRepository.existsById ( id )) {
-            Order orderDB = orderRepository.findById ( id ).get ( );
-            orderDB.setBeverages ( order.getBeverages ( ) );
-            orderDB.setSize ( order.getSize ( ) );
-            orderDB.setSugar ( order.getSugar ( ) );
-            orderRepository.save ( orderDB );
-        }
-        return "Order " + id + " not exist";
-
+    public Order updateOrder(UUID id, Order order) {
+        Order orderDB = findById(id);
+        orderDB.setBeverages(order.getBeverages());
+        orderDB.setSize(order.getSize());
+        orderDB.setSugar(order.getSugar());
+        return orderRepository.save(orderDB);
     }
 
     public boolean existOrder(UUID id) {
-        return orderRepository.existsById ( id );
+        try {
+            return orderRepository.existsById(id);
+        } catch (HttpClientErrorException ex) {
+            throw new ResourceNotFoundException(ex.getStatusCode(), ex.getStatusText(), ex.getResponseBodyAsByteArray(),
+                    Charset.defaultCharset());
+        }
```

Comment:
I'm not sure you need this method at all, you should use findById method and if the object doesn't exist it the DB you should throw an exception that's all